### ISSUE TITLE
amp.dev README.md should including instructions on installing pip for MacOS

### ISSUE THREAD

**Issue by CrystalOnScript** [ampproject/amp.dev#2888]
## 📖 Missing or out-of-date documentation

Downloading and running amp.dev is difficult on Mac, we should have instructions or link to proper documentation. Any other OS this might be an issue for?

**Comment by derekcecillewis** [comment]
With the [recent discovery](https://github.com/ampproject/amp.dev/issues/2790) that Grow doesn't support Windows, [realistically] only the various Linux distributions would remain. Seeing as how the `pip` command is unavailable on fresh installations of Debian 10 (and possibly other distributions), it would probably be helpful to link to [pip's official documentation](https://packaging.python.org/tutorials/installing-packages/#ensure-you-can-run-pip-from-the-command-line) for installation troubleshooting. There don't seem to be any noticeable stumbling blocks that aren't already covered by either the link above or the merged Brew-specific instructions.

---

After taking another look at the README, I noticed that further improvements are possible by eliminating parts of it that have become unnecessary since we added the below line.
https://github.com/ampproject/amp.dev/blob/d149de5016d78dd72e152b01c1a306744e363462/README.md#L19

For example, it doesn't seem necessary to specify operating systems in the following line seeing as how the intended audience for the instructions at this point are users of Unix-like OSes (no alternatives are specified nor are necessary).
https://github.com/ampproject/amp.dev/blob/d149de5016d78dd72e152b01c1a306744e363462/README.md#L25

---

It seems like this document could use a little refactoring before it reads correctly.

/to @CrystalOnScript Mind if I handle this issue?
